pkg/geo/geomfn/distance.go

```diff
@@ -887,16 +893,29 @@ func PointKindRelatesToPolygonKind(
 				return false, err
 			}
 			if intersects {
+				intersectsOnce = true
```

Comment:
if we're returning true anyway, can we avoid the `pointOuterLoop` and just return if it does not intersect on `PointPolygonCoveredBy`?

Reply:
Knowing that a point in the multipoint doesn't intersect with a particular polygon in the multipolygon is not enough to determine that the point doesn't intersect with the multipolygon, since it may intersect with a different polygon. So it's necessary to check each point against every polygon in the multipolygon until we either find an intersection (in which case we continue to the next point) or we exhaust all the polygons (in which case we return false, which is handled in the code after the inner for loop).

Reply:
Ok. I'm starting to think this is going to be a very confusing loop once we add Within and Contains. Wonder if we can simplify this somehow by splitting it into functions or the like. but we can get into it once we have to within / contains code in.